tools/harvest_source_import/remote_ckan/lib.py

```diff
@@ -0,0 +1,139 @@
+            error = f' - ERROR Response {response.status_code} {response.text}'
+            logger.error(error)
+            raise ValueError(error)
+
+        data = response.json()
+
+        if not data['success']:
+            error = 'ERROR searching harvest sources {}'.format(data['error'])
+            logger.error(error)
+            raise ValueError(error)
+
+        total = data['result']['count']
+        count = len(data['result']['results'])
+        if count == 0:
+            return
+        harvest_sources = data['result']['results']
+        logger.info(f'{count} ({total}) harvest sources found')
+
+        for hs in harvest_sources:
+            title = hs['title']
+            source_type = hs['source_type']  # datajosn, waf, etc
+            state = hs['state']
+            
+            logger.info(f'  [{source_type}] Harvest source: {title} [{state}]')
+            if state == 'active':
+                yield hs
+
+        # if the page is not full, is the last one
+        if count + 1 < page_size:
+            return
+
+        # get next page   
+        yield from self.list_harvest_sources(source_type=source_type, start=start + page_size, page_size=page_size)
+    
+    def get_request_headers(self, include_api_key=True):
+        headers = {'User-Agent': f'{self.user_agent}'}
+        if include_api_key:
+            headers['X-CKAN-API-Key'] = self.api_key
+        return headers
+
+    def create_harvest_source(self, data, owner_org_id):
+        """ create a harvest source (is just a CKAN dataset/package)
+            params:
+                data (dict): Harvest source dict
+                owner_org_id (str): Name or ID of the organization who owns this harvest source
+                
+            returns:
+                created (boolean):
+                status_code (int): request status code
+                error (str): None or error
+            """
+    
+        ckan_package = {
+            'name': data['name'],
+            'owner_org': owner_org_id,
+            'title': data['title'],
+            'url': data['url'],
+            'notes': data['notes'],
+            'source_type': data['source_type'],
+            'frequency': data['frequency'],
+            'config': data.get('config', '{}')
+        } 
+
+        # TODO should we create an organization? 
+        # TODO Check all other fields
+        # organization = ckan_package['organization']
+
+        package_create_url = f'{self.destination_url}/api/3/action/harvest_source_create'
+        headers = self.get_request_headers(include_api_key=True)
+
+        logger.info('Creating source URL {}'.format(ckan_package['title']))
+
+        try:
+            req = requests.post(package_create_url, data=ckan_package, headers=headers)
+        except Exception as e:
+            error = 'ERROR creating harvest source: {} [{}]'.format(e, ckan_package)
+            return False, 0, error
+
+        content = req.content
```

Comment:
Should this be in one of the `try` blocks in case there is no `req.content`

Reply:
On line 120?

Reply:
Moved error detection before try to read _content_ [here](https://github.com/GSA/catalog.data.gov/commit/8f622b6c8d1ed0298a6893cc76dcd299756e1516)